### DISCUSSION TITLE
Collection initializer clear feature

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/csharplang#6084] [Language Ideas]
Can we add a way to clear a collection before adding items to it using collection initializer in object initializer block?
```cs
var p = new PanelWithExistingChildren
{
  Controls = override { // override keyword triggers calling Clear() method (if doesn't exist, compiler error)
    new Button(),
  },
};
```

**Comment by saint4eva** [comment]
You could check out these #5671 and #5176

**Reply by TahirAhmadov** [reply]
I've seen both of those and they're great proposals, but I'm requesting a different type of tweak.

**Reply by saint4eva** [reply]
OKay. Cool

**Comment by HaloFour** [comment]
I don't like the idea of using a keyword here, especially `override` which means something very different in the language.  I believe that the described use case is very rare and typically unnecessary, so dedicating specific syntax to accomplish that (and only that) doesn't make a lot of sense.  I'd rather a more generic way to invoke methods within an initializer.
